src/msg/producer/writer/message_writer.go

```diff
@@ -303,10 +303,9 @@ func (w *messageWriterImpl) nextRetryNanos(writeTimes int, nowNanos int64) int64
 }
 
 func (w *messageWriterImpl) Ack(meta metadata) bool {
-	// acked, initNanos := w.acks.ack(meta)
-	acked, _ := w.acks.ack(meta)
+	acked, initNanos := w.acks.ack(meta)
 	if acked {
-		// w.m.messageConsumeLatency.Record(time.Duration(w.nowFn().UnixNano() - initNanos))
+		w.m.messageConsumeLatency.Record(time.Duration(w.nowFn().UnixNano() - initNanos))
```

Comment:
nit: maybe something like `w.m.messageConsumeLatency.Record(w.nowFn().Sub(time.Unix(0, initNanos))` ? Up to you which you think is nicer 👍

Reply:
ah, I see, probably will keep the existing way if that's cool with you.